src/replica/split/replica_split_manager.cpp

```diff
@@ -1101,5 +1103,20 @@ void replica_split_manager::child_handle_async_learn_error() // on child partiti
     _replica->_split_states.async_learn_task = nullptr;
 }
 
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica_split_manager::secondary_parent_handle_split(
+    const group_check_request &request,
+    /*out*/ group_check_response &response) // on secondary parent partition
+{
+    if (request.app.partition_count ==
+        _replica->_app_info.partition_count * 2) { // secondary update partition count
+        update_local_partition_count(request.app.partition_count);
+        parent_cleanup_split_context();
```

Comment:
Why we cleanup split context? The secondary split is proceeding. I guess `parent_cleanup_split_context` means initialization before split?

Reply:
`parent_cleanup_split_context` will be called when parent start to split and finish split.